kernel/aster-nix/src/syscall/mount.rs

```diff
@@ -0,0 +1,163 @@
+        do_new_mount(devname, fstype_addr, target_dentry)?;
+    }
+
+    Ok(SyscallReturn::Return(0))
+}
+
+fn do_reconfigure_mnt() -> Result<()> {
+    todo!()
+}
+
+fn do_remount() -> Result<()> {
+    todo!()
+}
+
+/// Bind a mount to a new location.
+fn do_loopback(old_name: CString, new_dentry: Arc<Dentry>, recursive: bool) -> Result<()> {
+    let current = current!();
+    let old_dentry = {
+        let old_name = old_name.to_string_lossy();
+        if old_name.is_empty() {
+            return_errno_with_message!(Errno::ENOENT, "old_name is empty");
+        }
+        let fs_path = FsPath::new(AT_FDCWD, old_name.as_ref())?;
+        current.fs().read().lookup(&fs_path)?
+    };
+
+    if old_dentry.type_() != InodeType::Dir {
+        return_errno_with_message!(Errno::ENOTDIR, "old_name must be directory");
+    };
+
+    let new_mount = old_dentry.do_loopback(recursive);
+    new_mount.graft_mount_tree(new_dentry.clone())?;
+    Ok(())
+}
+
+fn do_change_type() -> Result<()> {
+    todo!()
+}
+
+/// Move a mount from old location to new location.
+fn do_move_mount_old(old_name: CString, new_dentry: Arc<Dentry>) -> Result<()> {
+    let current = current!();
+    let old_dentry = {
+        let old_name = old_name.to_string_lossy();
+        if old_name.is_empty() {
+            return_errno_with_message!(Errno::ENOENT, "old_name is empty");
+        }
+        let fs_path = FsPath::new(AT_FDCWD, old_name.as_ref())?;
+        current.fs().read().lookup(&fs_path)?
+    };
+
+    if !old_dentry.is_root_of_mount() {
+        return_errno_with_message!(Errno::EINVAL, "old_name can not be moved");
+    };
+    if old_dentry.mount_node().parent().is_none() {
+        return_errno_with_message!(Errno::EINVAL, "old_name can not be moved");
+    }
+
+    old_dentry
+        .mount_node()
+        .graft_mount_tree(new_dentry.clone())?;
+
+    Ok(())
+}
+
+/// Mount a new filesystem.
+fn do_new_mount(devname: CString, fs_type: Vaddr, target_dentry: Arc<Dentry>) -> Result<()> {
+    let fs_type = read_cstring_from_user(fs_type, MAX_FILENAME_LEN)?;
+    if fs_type.is_empty() {
+        return_errno_with_message!(Errno::EINVAL, "fs_type is empty");
+    }
+    let fs_type = fs_type.to_str().unwrap();
+    let fs = crate::fs::get_fs(fs_type)?;
+    target_dentry.mount(fs)?;
+    Ok(())
+}
+
+bitflags! {
+    struct MountFlags: u32 {
+        const MS_RDONLY = 1;              /* Mount read-only */
+        const MS_NOSUID = 1<<1;           /* Ignore suid and sgid bits */
+        const MS_NODEV = 1<<2;            /* Disallow access to device special files */
+        const MS_NOEXEC = 1<<3;           /* Disallow program execution */
+        const MS_SYNCHRONOUS = 1<<4;      /* Writes are synced at once */
+        const MS_REMOUNT = 1<<5;          /* Alter flags of a mounted FS */
+        const MS_MANDLOCK = 1<<6;         /* Allow mandatory locks on an FS */
+        const MS_DIRSYNS = 1<<7;          /* Directory modifications are synchronous */
```

Comment:
Thank you for catching that mistake. Indeed, it should be `MS_DIRSYNC` instead of `MS_DIRSYNS`. I have reviewed and updated   other flags as well to ensure clarity and accuracy. Additionally, I noticed that `MS_VERBOSE` is no longer in use, so I have removed it from the list. Your vigilance is greatly appreciated, and it helps us maintain the quality and relevance of code .